zerver/tests/test_external.py

```diff
@@ -198,6 +198,36 @@ def assert_func(result: HttpResponse) -> None:
             finally:
                 remove_ratelimit_rule(1, 5, domain="create_realm_by_ip")
 
+    def test_find_accountrate_limiting(self) -> None:
+        def assert_func(result: HttpResponse) -> None:
+            self.assertEqual(result.status_code, 429)
+            self.assert_in_response("Rate limit exceeded.", result)
+
+        add_ratelimit_rule(1, 5, domain="find_account_by_ip")
```

Comment:
It'd probably be a better coding pattern to have the tests use a `with self.rate_limit_rule()...` pattern that does the try/finally for you; but that's a very reasonable follow-up refactor.

Reply:
Good idea, I pushed a new commit to do this cleanup